test/test-span-data.ts

```diff
@@ -14,177 +14,174 @@
  * limitations under the License.
  */
 
-'use strict';
+import * as assert from 'assert';
 
-import './override-gcp-metadata';
-import { cls } from '../src/cls';
-import { Constants } from '../src/constants';
+import {Constants, SpanDataType} from '../src/constants';
+import {BaseSpanData, ChildSpanData, RootSpanData} from '../src/span-data';
+import {Trace} from '../src/trace';
 import {TraceLabels} from '../src/trace-labels';
-import { RootSpanData, ChildSpanData } from '../src/span-data';
-import { Trace } from '../src/trace';
-import { traceWriter } from '../src/trace-writer';
-
-var assert = require('assert');
-var common = require('./plugins/common'/*.js*/);
-
-function createRootSpanData(name, traceId?, parentId?, skipFrames?) {
-  return new RootSpanData(
-    { projectId: '', traceId: traceId, spans: [] },
-    name,
-    parentId,
-    skipFrames
-  );
-}
-
-describe('SpanData', function() {
-  var agent;
-  before(function() {
-    agent = require('../..').start({
-      projectId: '0',
-      samplingRate: 0
-    });
+import {traceWriter, TraceWriter, TraceWriterConfig} from '../src/trace-writer';
+
+import {TestLogger} from './logger';
+import * as traceAgentModule from './trace';
+
+describe('SpanData', () => {
+  class CaptureSpanTraceWriter extends TraceWriter {
+    writeSpan(trace: Trace) {
+      capturedTrace = trace;
+    }
+  }
+  let capturedTrace: Trace;
+  let trace: Trace;
+
+  before(() => {
+    traceAgentModule.setTraceWriter(CaptureSpanTraceWriter);
+    traceWriter.create(
+        {
+          onUncaughtException: 'ignore',
+          maximumLabelValueSize: 16,
+          stackTraceLimit: 5
+        } as TraceWriterConfig,
+        new TestLogger());
   });
 
-  it('has correct default values', function() {
-    cls.get().runWithNewContext(function() {
-      var spanData = createRootSpanData('name', 1, 2);
-      assert.ok(spanData.trace);
-      assert.strictEqual(spanData.trace.traceId, 1);
-      assert.ok(spanData.span.spanId);
-      assert.strictEqual(spanData.span.name, 'name');
-    });
+  after(() => {
+    traceAgentModule.setTraceWriter(traceAgentModule.TestTraceWriter);
```

Comment:
This sets it to its original value. We're currently not running multiple test files in the same process but this restores original conditions from before any tests in this file were run.

Reply:
I see. So it's what `traceAgentModule` does, right? Its name is confusing to me because it reads like a real module, not a test module. Maybe name it 'testTraceAgentModule`?

I also realize that `setTraceWriter()` is a test-only function. Can it be named `setTraceWriterForTest()` explicitly?

Reply:
I changed `traceAgentModule` to `testTraceModule`... I can see why that was unclear. I don't think `setTraceWriterForTest` is needed though, since now the word `test` is already present on the object.

Reply:
BTW you didn't actually change `traceAgentModule` to `testTraceModule`...

Reply:
Oops, I forgot to push the changes... I'll open another PR

Also, point taken about `set*ForTest`. Will make that change as well